wicketd/src/artifacts.rs

```diff
@@ -39,6 +41,73 @@ struct ArtifactsWithPlan {
     plan: Option<UpdatePlan>,
 }
 
+/// The artifact server interface for wicketd.
+#[derive(Debug)]
+pub(crate) struct WicketdArtifactServer {
+    log: Logger,
+    store: WicketdArtifactStore,
+    ipr_artifact: IprArtifactServer,
```

Comment:
I didn't want to make this `Clone` unnecessarily, so I split up the server and store into separate components.